prowlarr/tag.go

```diff
@@ -0,0 +1,97 @@
+package prowlarr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"golift.io/starr"
+)
+
+// GetTags returns all the tags.
+func (p *Prowlarr) GetTags() ([]*starr.Tag, error) {
+	return p.GetTagsContext(context.Background())
+}
+
+// GetTagsContext returns all the tags.
+func (p *Prowlarr) GetTagsContext(ctx context.Context) ([]*starr.Tag, error) {
+	var tags []*starr.Tag
+
+	err := p.GetInto(ctx, "v1/tag", nil, &tags)
+	if err != nil {
+		return nil, fmt.Errorf("api.Get(tag): %w", err)
+	}
+
+	return tags, nil
+}
+
+// AddTag adds a tag or returns the ID for an existing tag.
+func (p *Prowlarr) AddTag(label string) (int, error) {
+	return p.AddTagContext(context.Background(), label)
+}
+
+// AddTagContext adds a tag or returns the ID for an existing tag.
+func (p *Prowlarr) AddTagContext(ctx context.Context, label string) (int, error) {
+	body, err := json.Marshal(&starr.Tag{Label: label, ID: 0})
+	if err != nil {
+		return 0, fmt.Errorf("json.Marshal(tag): %w", err)
+	}
+
+	var tag starr.Tag
+	if err = p.PostInto(ctx, "v1/tag", nil, body, &tag); err != nil {
+		return tag.ID, fmt.Errorf("api.Post(tag): %w", err)
+	}
+
+	return tag.ID, nil
+}
+
+// UpdateTag updates the label for a tag.
+func (p *Prowlarr) UpdateTag(tagID int, label string) (int, error) {
```

Comment:
Is Prowlarr the only one with update?

Reply:
no, the others already had update, I just changed it to add the ID. It works even without it, but in all api docs but sonarr's the ID is in the definition (e.g. https://radarr.video/docs/api/#/Tag/put-tag-id)

Reply:
Gotcha, cool. I think something I wrote uses update tag, and it's been around a while so I never made it back there to the rest of the methods.